File: dsa-practice/maths/problem7.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    cin>>n;
    if(n==1){
        cout<<0;
    }
    else if(n==2){
        cout<<0<<" "<<1;
    }
    else{
        int fibo,a=0,b=1,i=3;
        cout<<a<<" "<<b<<" ";
        while(i<=n){
            fibo = a+b;
            cout<<fibo<<" ";
            a = b;
            b = fibo;
            i++;
        }
        cout<<endl;

    }
    cout<<endl;
    return 0;
}

File: dsa-practice/maths/problem7.js
const fibo = (num)=>{
    if(num===1){
        return 0
    }
    else if(num===2){
        return "0 1"
    }
    else{
        let arr = [0,1]
        let a=0,b=1,i=3;
        while(i<=num){
            let fibo = a+b;
            arr.push(fibo);
            a = b;
            b = fibo;
            i++;
        }
        return arr;
    }
}
console.log(fibo(10))